app/lib/search/search_service.dart

```diff
@@ -108,6 +111,7 @@ class PackageDocument {
       doNotAdvertise: doNotAdvertise,
       supportsOnlyLegacySdk: supportsOnlyLegacySdk,
       platforms: platforms?.map(internFn)?.toList(),
+      tags: tags?.map(internFn)?.toList(),
```

Comment:
If `tags` can be `null` we should documented as noted above, and leave a TODO to remove the `null` handling when `tags` cannot be null anymore.